### PR TITLE
docs: refactor referer-restriction.md

### PR DESCRIPTION
**Pull request by lunarwhite**
### Description

update document under `docs/zh/latest/plugins/` and `docs/en/latest/plugins/`

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/zh/latest/plugins/referer-restriction.md [hunk 1]
```diff
@@ -23,23 +29,26 @@ title: referer-restriction
...
+| 参数名    | 类型          | 必选项 | 默认值 | 有效值 | 描述                             |
 | --------- | ------------- | ------ | ------ | ------ | -------------------------------- |
-| whitelist | array[string] | 可选    |         |       | 白名单域名列表。域名开头可以用'*'作为通配符。 |
-| blacklist | array[string] | 可选    |         |       | 黑名单域名列表。域名开头可以用'*'作为通配符。 |
-| message | string | 可选    | Your referer host is not allowed | [1, 1024] | 在未允许访问的情况下返回的信息。 |
-| bypass_missing  | boolean       | 可选    | false   |       | 当 Referer 不存在或格式有误时，是否绕过检查。 |
+| whitelist | array[string] | 否    |         |       | 白名单域名列表。域名开头可以用 `*` 作为通配符。 |
+| blacklist | array[string] | 否    |         |       | 黑名单域名列表。域名开头可以用 `*` 作为通配符。 |
+| message | string | 否    | "Your referer host is not allowed" | [1, 1024] | 在未允许访问的情况下返回的信息。 |
+| bypass_missing  | boolean       | 否    | false   |       | 当 `Referer` 请求头不存在或格式有误时，是否绕过检查。如果设置该选项为 `true`，那么将绕过检查。 |
```
Comment (hf400159) [inline docs/zh/latest/plugins/referer-restriction.md:41]:
```suggestion
| bypass_missing  | boolean       | 否    | false   |       | 当设置为 `true` 时，如果 `Referer` 请求头不存在或格式有误时，将绕过检查。 |
```

File: docs/zh/latest/plugins/referer-restriction.md [hunk 2]
```diff
@@ -23,23 +29,26 @@ title: referer-restriction
...
 ## 属性
 
-| 参数名    | 类型          | 可选项 | 默认值 | 有效值 | 描述                             |
+| 参数名    | 类型          | 必选项 | 默认值 | 有效值 | 描述                             |
 | --------- | ------------- | ------ | ------ | ------ | -------------------------------- |
-| whitelist | array[string] | 可选    |         |       | 白名单域名列表。域名开头可以用'*'作为通配符。 |
-| blacklist | array[string] | 可选    |         |       | 黑名单域名列表。域名开头可以用'*'作为通配符。 |
-| message | string | 可选    | Your referer host is not allowed | [1, 1024] | 在未允许访问的情况下返回的信息。 |
-| bypass_missing  | boolean       | 可选    | false   |       | 当 Referer 不存在或格式有误时，是否绕过检查。 |
+| whitelist | array[string] | 否    |         |       | 白名单域名列表。域名开头可以用 `*` 作为通配符。 |
```
Comment (hf400159) [inline docs/zh/latest/plugins/referer-restriction.md:38]:
```suggestion
| whitelist | array[string] | 否    |         |       | 白名单域名列表。域名开头可以用 `*` 作为通配符。`whitelist` 和 `blacklist` 必须设置一个。 |
```

File: docs/zh/latest/plugins/referer-restriction.md [hunk 3]
```diff
@@ -23,23 +29,26 @@ title: referer-restriction
...
+| blacklist | array[string] | 否    |         |       | 黑名单域名列表。域名开头可以用 `*` 作为通配符。 |
```
Comment (hf400159) [inline docs/zh/latest/plugins/referer-restriction.md:39]:
```suggestion
| blacklist | array[string] | 否    |         |       | 黑名单域名列表。域名开头可以用 `*` 作为通配符。`whitelist` 和 `blacklist` 必须设置一个。|
```

File: docs/zh/latest/plugins/referer-restriction.md (hunk 1 above)
Comment (hf400159) [reply docs/zh/latest/plugins/referer-restriction.md:41]: If set to false, what happens when the `Referer` request header does not exist?

File: docs/zh/latest/plugins/referer-restriction.md [hunk 4]
```diff
@@ -65,37 +74,52 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
 
 ## 测试插件
 
-带 `Referer: http://xx.com/x` 请求：
+通过上述命令启用插件后，你可以将请求设置为 `Referer: http://xx.com/x` 来测试插件：
```
Comment (hf400159) [inline docs/zh/latest/plugins/referer-restriction.md:77]:
```suggestion
通过上述命令启用插件后，你可以在请求中添加 `Referer: http://xx.com/x` 测试插件：
```

File: docs/zh/latest/plugins/referer-restriction.md [hunk 5]
```diff
@@ -65,37 +74,52 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+
+```shell
+curl http://127.0.0.1:9080/index.html -H 'Referer: http://xx.com/x'
+```
+
+返回的 HTTP 响应头中带有 `200` 状态码，代表访问成功：
```
Comment (hf400159) [inline docs/zh/latest/plugins/referer-restriction.md:83]:
```suggestion
返回的 HTTP 响应头中带有 `200` 状态码则表示访问成功：
```

File: docs/zh/latest/plugins/referer-restriction.md [hunk 6]
```diff
@@ -65,37 +74,52 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
 ```shell
-$ curl http://127.0.0.1:9080/index.html
 HTTP/1.1 200 OK
 ...
 ```
 
 ## 禁用插件
 
-当你想去掉 `referer-restriction` 插件的时候，很简单，在插件的配置中把对应的 json 配置删除即可，无须重启服务，即刻生效：
+当你需要禁用 `referer-restriction` 插件时，可以通过以下命令删除相应的 JSON 配置，APISIX 将会自动重新加载相关配置，无需重启服务：
```
Comment (hf400159) [inline docs/zh/latest/plugins/referer-restriction.md:119]:
```suggestion
当你需要禁用该插件时，可以通过以下命令删除相应的 JSON 配置，APISIX 将会自动重新加载相关配置，无需重启服务：
```

File: docs/en/latest/plugins/referer-restriction.md [hunk 7]
```diff
@@ -2,7 +2,7 @@
 title: referer-restriction
 keywords:
   - APISIX
-  - Plugin
+  - API Gateway
   - Referer restriction
   - referer-restriction
 description: This document contains information about the Apache APISIX referer-restriction Plugin.
```
Comment (hf400159) [inline docs/en/latest/plugins/referer-restriction.md:8]: Please refer to the Chinese modification, keep the characters between 130 and 150.

**Review by hf400159 — APPROVED** [review]

File: docs/zh/latest/plugins/referer-restriction.md [hunk 8]
```diff
@@ -23,23 +29,26 @@ title: referer-restriction
 
 ## 描述
 
-`referer-restriction` 插件可以根据 Referer 请求头限制访问。
+`referer-restriction` 插件允许用户将 `Referer` 请求头列入白名单或黑名单来限制对服务或路由的访问。
 
 ## 属性
 
-| 参数名    | 类型          | 可选项 | 默认值 | 有效值 | 描述                             |
+| 参数名    | 类型          | 必选项 | 默认值 | 有效值 | 描述                             |
```
Comment (hf400159) [inline docs/zh/latest/plugins/referer-restriction.md:36]:
```suggestion
| 名称    | 类型          | 必选项 | 默认值 | 有效值 | 描述                             |
```

File: docs/zh/latest/plugins/referer-restriction.md [hunk 9]
```diff
@@ -65,37 +74,52 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
 
 ```shell
-$ curl http://127.0.0.1:9080/index.html -H 'Referer: http://yy.com/x'
 HTTP/1.1 403 Forbidden
 ...
 {"message":"Your referer host is not allowed"}
 ```
 
-不带 `Referer` 请求：
+因为启用插件时将属性 `bypass_missing` 设置为 `true`，所以未指定 `Refer` 请求头的请求将跳过检查：
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/referer-restriction.md:104]:
```suggestion
因为启用插件时会将属性 `bypass_missing` 设置为 `true`，所以未指定 `Refer` 请求头的请求将跳过检查：
```

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**